src/stateStream.spec.ts

```diff
@@ -0,0 +1,134 @@
+import { reducer, actionCreator } from 'rxbeach';
+import untypedTest from 'ava';
+import { marbles } from 'rxjs-marbles/ava';
+import sinon from 'sinon';
+import { stateStream, stateStreamWithErrorHandling } from 'rxbeach/stateStream';
+import { of } from 'rxjs';
+import { stubRethrowErrorGlobally } from 'rxbeach/internal/testUtils';
+
+const incrementOne = actionCreator('[increment] one');
+const decrement = actionCreator('[increment] decrement');
+const incrementMany = actionCreator<number>('[increment] many');
+
+const incrementOneHandler = sinon.spy((accumulator: number) => accumulator + 1);
+const handleOne = reducer(incrementOne, incrementOneHandler);
+const handleMany = reducer(
+  incrementMany,
+  (accumulator: number, increment) => accumulator + increment
+);
+const test = stubRethrowErrorGlobally(untypedTest);
+
+const actions = {
+  '1': incrementOne(),
+  '2': incrementMany(2),
+  d: decrement(),
+};
+const words = {
+  a: '1',
+  b: '12',
+};
+const numbers = {
+  '1': 1,
+  '2': 2,
+  '3': 3,
+  '4': 4,
+  '5': 5,
+  '6': 6,
+  '7': 7,
+};
+
+test(
+  'stateStream subscribes action$',
+  marbles(m => {
+    const defaultState = 3;
+    const action$ = m.hot('', actions);
+    const state$ = stateStream('testStream', action$, defaultState, []);
+    state$.subscribe();
+    m.expect(action$).toHaveSubscriptions(['^']);
+  })
+);
+
+test(
+  'stateStream should emit initial state synchronously upon subscription',
+  marbles(m => {
+    const defaultState = 3;
+    const action$ = m.hot('');
```

Comment:
```suggestion
    const action$ = m.hot('  -');
```
To explicitly indicate it doesn't emit anything in the time frame the other stream emits `3`